`index.php/204` throttle causes app to believe there's no connection

### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Take a picture ( ~ 5 MB)
2. Wait for instant upload
3. Retry upload

### Expected behaviour

Picture uploaded properly

### Actual behaviour

Always shows connection error in the uploads list.
Happens for 2 Samsung mobile phones.

### Android version

12

### Device brand and model

Samsung Galaxy A52s 5G

### Stock or custom OS?

Stock

### Nextcloud android app version

3.22.3

### Nextcloud server version

25.0.1 RC1

### Using a reverse proxy?

No

### Android logs

TODO: if not reproducible on your side I can try and grab logs (logcat is not working I need time to setup/debug)

### Server error logs

```bash
None, the upload connection doesn't even reach the server!
```


### Additional information

Note: I used to have the issue with conflicts on upload on two mobile phones https://github.com/nextcloud/android/issues/7811 and after upgrading to 3.22.3 the above issue is happening

I've checked the access.log and during upload the client does not even reach the server!

However, regular browsing through folders works, I can see a lot of capability requests and PROPFINDs.
It's only uploads that fail to connect.

Is upload using another kind of connection manager ? @tobiasKaminsky @AlvaroBrey

okay, I got something from logcat:
```
11-06 11:40:53.066 16148 16250 E GetMethod: Error executing method
11-06 11:40:53.066 16148 16250 E GetMethod: java.io.IOException: unexpected end of stream on https://cloud.vincentpetry.net/...
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:203)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:111)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:93)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:65)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:205)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.connection.RealCall.execute(RealCall.kt:158)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:175)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.nextcloud.client.network.ConnectivityServiceImpl.isInternetWalled(ConnectivityServiceImpl.java:81)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.owncloud.android.operations.UploadFileOperation.checkConditions(UploadFileOperation.java:728)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.owncloud.android.operations.UploadFileOperation.normalUpload(UploadFileOperation.java:747)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.owncloud.android.operations.UploadFileOperation.run(UploadFileOperation.java:442)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:205)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.nextcloud.client.jobs.FilesUploadWorker.upload(FilesUploadWorker.kt:155)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.nextcloud.client.jobs.FilesUploadWorker.handlePendingUploads(FilesUploadWorker.kt:103)
11-06 11:40:53.066 16148 16250 E GetMethod:     at com.nextcloud.client.jobs.FilesUploadWorker.doWork(FilesUploadWorker.kt:86)
11-06 11:40:53.066 16148 16250 E GetMethod:     at androidx.work.Worker$1.run(Worker.java:86)
11-06 11:40:53.066 16148 16250 E GetMethod:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-06 11:40:53.066 16148 16250 E GetMethod:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-06 11:40:53.066 16148 16250 E GetMethod:     at java.lang.Thread.run(Thread.java:920)
11-06 11:40:53.066 16148 16250 E GetMethod: Caused by: java.io.EOFException: \n not found: limit=0 content=…
11-06 11:40:53.066 16148 16250 E GetMethod:     at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:332)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29)
11-06 11:40:53.066 16148 16250 E GetMethod:     at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178)
11-06 11:40:53.066 16148 16250 E GetMethod:     ... 26 more
11-06 11:40:53.066 16148 16250 E UploadFileOperation: Upload of /storage/413E-1C03/DCIM/Camera/20221101_122043.jpg to /general/petry/pics/umstiegstation/20221101_122043.jpg: No network connection
11-06 11:40:53.067 16148 16250 D UploadsStorageManager: updateDatabaseUploadResult uploadResult: RemoteOperationResult{mSuccess=false, mHttpCode=-1, mHttpPhrase='null', mException=null, mCode=NO_NETWORK_CONNECTION, message='null', getLogMessage='No network connection'} upload: com.owncloud.android.operations.UploadFileOperation@6afafac
11-06 11:40:53.072 16148 16250 V UploadsStorageManager: Updating /storage/413E-1C03/DCIM/Camera/20221101_122043.jpg with status:UPLOAD_FAILED and result:NETWORK_CONNECTION (old:/storage/413E-1C03/DCIM/Camera/20221101_122043.jpg status:UPLOAD_IN_PROGRESS result:-1)
11-06 11:40:53.072 16148 16250 V UploadsStorageManager: Updating /storage/413E-1C03/DCIM/Camera/20221101_122043.jpg with status=UPLOAD_FAILED
```

--- a/app/src/androidTest/java/com/nextcloud/client/network/ConnectivityServiceImplIT.kt
+++ b/app/src/androidTest/java/com/nextcloud/client/network/ConnectivityServiceImplIT.kt
@@ -24,8 +24,8 @@ package com.nextcloud.client.network
 import android.accounts.AccountManager
 import android.content.Context
 import android.net.ConnectivityManager
-import android.os.Build
 import com.nextcloud.client.account.UserAccountManagerImpl
+import com.nextcloud.client.core.ClockImpl
 import com.nextcloud.client.network.ConnectivityServiceImpl.GetRequestBuilder
 import com.owncloud.android.AbstractOnServerIT
 import org.junit.Assert.assertFalse
@@ -40,13 +40,14 @@ class ConnectivityServiceImplIT : AbstractOnServerIT() {
         val userAccountManager = UserAccountManagerImpl(targetContext, accountManager)
         val clientFactory = ClientFactoryImpl(targetContext)
         val requestBuilder = GetRequestBuilder()
+        val walledCheckCache = WalledCheckCache(ClockImpl())
 
         val sut = ConnectivityServiceImpl(
             connectivityManager,
             userAccountManager,
             clientFactory,
             requestBuilder,
-            Build.VERSION.SDK_INT
+            walledCheckCache
         )
 
         assertTrue(sut.connectivity.isConnected)

--- a/app/src/main/java/com/nextcloud/client/network/ConnectivityServiceImpl.java
+++ b/app/src/main/java/com/nextcloud/client/network/ConnectivityServiceImpl.java
@@ -39,11 +39,14 @@ import kotlin.jvm.functions.Function1;
 class ConnectivityServiceImpl implements ConnectivityService {
 
     private static final String TAG = "ConnectivityServiceImpl";
+    private static final String CONNECTIVITY_CHECK_ROUTE = "/index.php/204";
+
     private final ConnectivityManager platformConnectivityManager;
     private final UserAccountManager accountManager;
     private final ClientFactory clientFactory;
     private final GetRequestBuilder requestBuilder;
-    private final int sdkVersion;
+    private final WalledCheckCache walledCheckCache;
+
 
     static class GetRequestBuilder implements Function1<String, GetMethod> {
         @Override
@@ -56,37 +59,49 @@ class ConnectivityServiceImpl implements ConnectivityService {
                             UserAccountManager accountManager,
                             ClientFactory clientFactory,
                             GetRequestBuilder requestBuilder,
-                            int sdkVersion) {
+                            final WalledCheckCache walledCheckCache) {
         this.platformConnectivityManager = platformConnectivityManager;
         this.accountManager = accountManager;
         this.clientFactory = clientFactory;
         this.requestBuilder = requestBuilder;
-        this.sdkVersion = sdkVersion;
+        this.walledCheckCache = walledCheckCache;
     }
 
     @Override
     public boolean isInternetWalled() {
-        Connectivity c = getConnectivity();
-        if (c.isConnected() && c.isWifi() && !c.isMetered()) {
+        final Boolean cachedValue = walledCheckCache.getValue();
+        if (cachedValue != null) {
+            return cachedValue;
+        } else {
+            boolean result;
+            Connectivity c = getConnectivity();
+            if (c.isConnected() && c.isWifi() && !c.isMetered()) {
 
-            Server server = accountManager.getUser().getServer();
-            String baseServerAddress = server.getUri().toString();
-            if (baseServerAddress.isEmpty()) {
-                return true;
+                Server server = accountManager.getUser().getServer();
+                String baseServerAddress = server.getUri().toString();
+                if (baseServerAddress.isEmpty()) {
+                    result = true;
+                } else {
+
+                    GetMethod get = requestBuilder.invoke(baseServerAddress + CONNECTIVITY_CHECK_ROUTE);
+                    PlainClient client = clientFactory.createPlainClient();
+
+                    int status = get.execute(client);
+
+                    // Content-Length is not available when using chunked transfer encoding, so check for -1 as well
+                    result = !(status == HttpStatus.SC_NO_CONTENT && get.getResponseContentLength() <= 0);
+                    get.releaseConnection();
+                    if (result) {
+                        Log_OC.w(TAG, "isInternetWalled(): Failed to GET " + CONNECTIVITY_CHECK_ROUTE + "," +
+                            " assuming connectivity is impaired");
+                    }
+                }
+            } else {
+                result = !c.isConnected();
             }
 
-            GetMethod get = requestBuilder.invoke(baseServerAddress + "/index.php/204");
-            PlainClient client = clientFactory.createPlainClient();
-
-            int status = get.execute(client);
-
-            // Content-Length is not available when using chunked transfer encoding, so check for -1 as well
-            boolean result = !(status == HttpStatus.SC_NO_CONTENT && get.getResponseContentLength() <= 0);
-            get.releaseConnection();
-
+            walledCheckCache.setValue(result);
             return result;
-        } else {
-            return !c.isConnected();
         }
     }
 

--- a/app/src/main/java/com/nextcloud/client/network/NetworkModule.java
+++ b/app/src/main/java/com/nextcloud/client/network/NetworkModule.java
@@ -37,12 +37,13 @@ public class NetworkModule {
     @Provides
     ConnectivityService connectivityService(ConnectivityManager connectivityManager,
                                             UserAccountManager accountManager,
-                                            ClientFactory clientFactory) {
+                                            ClientFactory clientFactory,
+                                            WalledCheckCache walledCheckCache) {
         return new ConnectivityServiceImpl(connectivityManager,
                                            accountManager,
                                            clientFactory,
                                            new ConnectivityServiceImpl.GetRequestBuilder(),
-                                           Build.VERSION.SDK_INT
+                                           walledCheckCache
         );
     }
 

--- a/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
+++ b/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
@@ -36,7 +36,7 @@ class WalledCheckCache @Inject constructor(private val clock: Clock) {
         return when (val timestamp = cachedEntry?.first) {
             null -> true
             else -> {
-                val diff = clock.millisSinceBoot - timestamp
+                val diff = clock.currentTime - timestamp
                 diff >= CACHE_TIME_MS
             }
         }
@@ -44,7 +44,7 @@ class WalledCheckCache @Inject constructor(private val clock: Clock) {
 
     @Synchronized
     fun setValue(isWalled: Boolean) {
-        this.cachedEntry = Pair(clock.millisSinceBoot, isWalled)
+        this.cachedEntry = Pair(clock.currentTime, isWalled)
     }
 
     @Synchronized

--- a/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
+++ b/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
@@ -29,11 +29,11 @@ import javax.inject.Singleton
 @Singleton
 class WalledCheckCache @Inject constructor(private val clock: Clock) {
 
-    private var value: Pair<Long, Boolean>? = null
+    private var cachedEntry: Pair<Long, Boolean>? = null
 
     @Synchronized
     fun isExpired(): Boolean {
-        return when (val timestamp = value?.first) {
+        return when (val timestamp = cachedEntry?.first) {
             null -> true
             else -> {
                 val diff = clock.millisSinceBoot - timestamp
@@ -43,21 +43,21 @@ class WalledCheckCache @Inject constructor(private val clock: Clock) {
     }
 
     @Synchronized
-    fun setValue(value: Boolean) {
-        this.value = Pair(clock.millisSinceBoot, value)
+    fun setValue(isWalled: Boolean) {
+        this.cachedEntry = Pair(clock.millisSinceBoot, isWalled)
     }
 
     @Synchronized
     fun getValue(): Boolean? {
         return when (isExpired()) {
             true -> null
-            else -> value?.second
+            else -> cachedEntry?.second
         }
     }
 
     @Synchronized
     fun clear() {
-        value = null
+        cachedEntry = null
     }
 
     companion object {

--- a/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
+++ b/app/src/main/java/com/nextcloud/client/network/WalledCheckCache.kt
@@ -1,0 +1,67 @@
+/*
+ * Nextcloud Android client application
+ *
+ *  @author Álvaro Brey
+ *  Copyright (C) 2022 Álvaro Brey
+ *  Copyright (C) 2022 Nextcloud GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.nextcloud.client.network
+
+import com.nextcloud.client.core.Clock
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class WalledCheckCache @Inject constructor(private val clock: Clock) {
+
+    private var value: Pair<Long, Boolean>? = null
+
+    @Synchronized
+    fun isExpired(): Boolean {
+        return when (val timestamp = value?.first) {
+            null -> true
+            else -> {
+                val diff = clock.millisSinceBoot - timestamp
+                diff >= CACHE_TIME_MS
+            }
+        }
+    }
+
+    @Synchronized
+    fun setValue(value: Boolean) {
+        this.value = Pair(clock.millisSinceBoot, value)
+    }
+
+    @Synchronized
+    fun getValue(): Boolean? {
+        return when (isExpired()) {
+            true -> null
+            else -> value?.second
+        }
+    }
+
+    @Synchronized
+    fun clear() {
+        value = null
+    }
+
+    companion object {
+        // 10 minutes
+        private const val CACHE_TIME_MS = 10 * 60 * 1000
+    }
+}

--- a/app/src/main/java/com/owncloud/android/MainApp.java
+++ b/app/src/main/java/com/owncloud/android/MainApp.java
@@ -54,6 +54,7 @@ import com.nextcloud.client.logger.LegacyLoggerAdapter;
 import com.nextcloud.client.logger.Logger;
 import com.nextcloud.client.migrations.MigrationsManager;
 import com.nextcloud.client.network.ConnectivityService;
+import com.nextcloud.client.network.WalledCheckCache;
 import com.nextcloud.client.onboarding.OnboardingService;
 import com.nextcloud.client.preferences.AppPreferences;
 import com.nextcloud.client.preferences.AppPreferencesImpl;
@@ -177,6 +178,8 @@ public class MainApp extends MultiDexApplication implements HasAndroidInjector {
 
     @Inject
     PassCodeManager passCodeManager;
+
+    @Inject WalledCheckCache walledCheckCache;
 
     // workaround because injection is initialized on onAttachBaseContext
     // and getApplicationContext is null at that point, which crashes when getting current user
@@ -326,7 +329,8 @@ public class MainApp extends MultiDexApplication implements HasAndroidInjector {
                            powerManagementService,
                            backgroundJobManager,
                            clock,
-                           viewThemeUtils);
+                           viewThemeUtils,
+                           walledCheckCache);
         initContactsBackup(accountManager, backgroundJobManager);
         notificationChannels();
 
@@ -506,8 +510,8 @@ public class MainApp extends MultiDexApplication implements HasAndroidInjector {
         final PowerManagementService powerManagementService,
         final BackgroundJobManager backgroundJobManager,
         final Clock clock,
-        final ViewThemeUtils viewThemeUtils
-                                         ) {
+        final ViewThemeUtils viewThemeUtils,
+        final WalledCheckCache walledCheckCache) {
         updateToAutoUpload();
         cleanOldEntries(clock);
         updateAutoUploadEntries(clock);
@@ -537,7 +541,8 @@ public class MainApp extends MultiDexApplication implements HasAndroidInjector {
         ReceiversHelper.registerNetworkChangeReceiver(uploadsStorageManager,
                                                       accountManager,
                                                       connectivityService,
-                                                      powerManagementService);
+                                                      powerManagementService,
+                                                      walledCheckCache);
 
         ReceiversHelper.registerPowerChangeReceiver(uploadsStorageManager,
                                                     accountManager,

--- a/app/src/main/java/com/owncloud/android/files/BootupBroadcastReceiver.java
+++ b/app/src/main/java/com/owncloud/android/files/BootupBroadcastReceiver.java
@@ -32,6 +32,7 @@ import com.nextcloud.client.core.Clock;
 import com.nextcloud.client.device.PowerManagementService;
 import com.nextcloud.client.jobs.BackgroundJobManager;
 import com.nextcloud.client.network.ConnectivityService;
+import com.nextcloud.client.network.WalledCheckCache;
 import com.nextcloud.client.preferences.AppPreferences;
 import com.owncloud.android.MainApp;
 import com.owncloud.android.datamodel.UploadsStorageManager;
@@ -59,6 +60,7 @@ public class BootupBroadcastReceiver extends BroadcastReceiver {
     @Inject BackgroundJobManager backgroundJobManager;
     @Inject Clock clock;
     @Inject ViewThemeUtils viewThemeUtils;
+    @Inject WalledCheckCache walledCheckCache;
 
     /**
      * Receives broadcast intent reporting that the system was just boot up. *
@@ -78,7 +80,8 @@ public class BootupBroadcastReceiver extends BroadcastReceiver {
                                        powerManagementService,
                                        backgroundJobManager,
                                        clock,
-                                       viewThemeUtils);
+                                       viewThemeUtils,
+                                       walledCheckCache);
             MainApp.initContactsBackup(accountManager, backgroundJobManager);
         } else {
             Log_OC.d(TAG, "Getting wrong intent: " + intent.getAction());

--- a/app/src/main/java/com/owncloud/android/utils/ReceiversHelper.java
+++ b/app/src/main/java/com/owncloud/android/utils/ReceiversHelper.java
@@ -30,6 +30,7 @@ import android.content.IntentFilter;
 import com.nextcloud.client.account.UserAccountManager;
 import com.nextcloud.client.device.PowerManagementService;
 import com.nextcloud.client.network.ConnectivityService;
+import com.nextcloud.client.network.WalledCheckCache;
 import com.nextcloud.common.DNSCache;
 import com.owncloud.android.MainApp;
 import com.owncloud.android.datamodel.UploadsStorageManager;
@@ -46,7 +47,8 @@ public final class ReceiversHelper {
     public static void registerNetworkChangeReceiver(final UploadsStorageManager uploadsStorageManager,
                                                      final UserAccountManager accountManager,
                                                      final ConnectivityService connectivityService,
-                                                     final PowerManagementService powerManagementService) {
+                                                     final PowerManagementService powerManagementService,
+                                                     final WalledCheckCache walledCheckCache) {
         Context context = MainApp.getAppContext();
 
         IntentFilter intentFilter = new IntentFilter();
@@ -57,6 +59,7 @@ public final class ReceiversHelper {
             @Override
             public void onReceive(Context context, Intent intent) {
                 DNSCache.clear();
+                walledCheckCache.clear();
                 if (connectivityService.getConnectivity().isConnected()) {
                     FilesSyncHelper.restartJobsIfNeeded(uploadsStorageManager,
                                                         accountManager,

--- a/app/src/test/java/com/nextcloud/client/network/ConnectivityServiceTest.kt
+++ b/app/src/test/java/com/nextcloud/client/network/ConnectivityServiceTest.kt
@@ -23,7 +23,6 @@ import android.net.ConnectivityManager
 import android.net.Network
 import android.net.NetworkCapabilities
 import android.net.NetworkInfo
-import android.os.Build
 import com.nextcloud.client.account.Server
 import com.nextcloud.client.account.User
 import com.nextcloud.client.account.UserAccountManager
@@ -98,6 +97,9 @@ class ConnectivityServiceTest {
         lateinit var network: Network
 
         @Mock
+        lateinit var walledCheckCache: WalledCheckCache
+
+        @Mock
         lateinit var networkCapabilities: NetworkCapabilities
 
         @Mock
@@ -120,7 +122,7 @@ class ConnectivityServiceTest {
                 accountManager,
                 clientFactory,
                 requestBuilder,
-                Build.VERSION_CODES.Q
+                walledCheckCache
             )
 
             whenever(networkCapabilities.hasCapability(eq(NetworkCapabilities.NET_CAPABILITY_NOT_RESTRICTED)))
@@ -133,6 +135,7 @@ class ConnectivityServiceTest {
             whenever(clientFactory.createPlainClient()).thenReturn(client)
             whenever(user.server).thenReturn(newServer)
             whenever(accountManager.user).thenReturn(user)
+            whenever(walledCheckCache.getValue()).thenReturn(null)
         }
     }
 

